src/coreclr/tools/Common/TypeSystem/Ecma/EcmaType.TypeEquivalence.cs

```diff
@@ -80,12 +80,12 @@ private TypeIdentifierData ComputeTypeIdentifierData()
                     if (this.HasCustomAttribute("System.Runtime.InteropServices", "ImportedFromTypeLibAttribute") || this.HasCustomAttribute("System.Runtime.InteropServices", "PrimaryInteropAssemblyAttribute"))
                     {
                         // This type has a TypeIdentifier attribute if it has an appropriate shape to be considered type equivalent
-                    }
-
-                    if (!TypeHasCharacteristicsRequiredToBeTypeEquivalent)
-                        return null;
+                        if (!TypeHasCharacteristicsRequiredToBeTypeEquivalent)
+                            return null;
 
-                    _data = ComputeTypeIdentifierFromGuids();
+                        _data = ComputeTypeIdentifierFromGuids();
+                        return _data;
```

Comment:
nit: I think this function (and `ComputeTypeIdentifierFromGuids`) should not be assigning to `_data` directly.  They should just return a `TypeIdentifierData` or null.  The caching getter should do the atomic assignment to `_data`

Reply:
This is what I get for working on a PR over 2-3 years. I forget stuff. I don't think it's really a functional problem, but only having 1 writer was the intention here.